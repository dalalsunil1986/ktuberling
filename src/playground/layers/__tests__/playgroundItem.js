import PlaygroundItem from '../playgroundItem';

// L.Polygon.extend
jest.mock('leaflet', () => ({
  Polygon: {
    extend: jest.fn(obj => (...args) => {
      const extendedObj = {
        ...obj,
      };

      extendedObj.initialize(...args);
      return extendedObj;
    }),
  },
}));

jest.mock('extra-log', () => ({ msg: jest.fn() }));

describe('App', () => {
  const node = { _name: 'node' };


  it('Init', () => {
    const item = new PlaygroundItem(node);

    expect(item).toMatchSnapshot();
  });


  it('remove', () => {
    const item = new PlaygroundItem(node);

    item.remove();

    expect(item).toMatchSnapshot();
  });


  it('onAdd', () => {
    const map = { _name: 'map' };
    const item = new PlaygroundItem(node);
    item.onAdd(map);
  });


  it('onRemove', () => {
    const map = { _name: 'map' };
    const item = new PlaygroundItem(node);
    item.onRemove(map);
  });
});
