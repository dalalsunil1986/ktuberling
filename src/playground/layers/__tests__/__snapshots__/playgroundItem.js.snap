// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Init 1`] = `
Object {
  "_node": Object {
    "_name": "node",
  },
  "initialize": [Function],
  "onAdd": [Function],
  "onRemove": [Function],
  "options": Object {
    "className": "playground-item",
    "highlightColor": "grey",
  },
  "remove": [Function],
}
`;

exports[`App remove 1`] = `
Object {
  "_node": undefined,
  "initialize": [Function],
  "onAdd": [Function],
  "onRemove": [Function],
  "options": Object {
    "className": "playground-item",
    "highlightColor": "grey",
  },
  "remove": [Function],
}
`;
